Source/Eto.WinForms/Forms/WindowsControl.cs

```diff
@@ -91,6 +93,49 @@ public abstract class WindowsControl<TControl, TWidget, TCallback> : WidgetHandl
 		where TWidget : Control
 		where TCallback : Control.ICallback
 	{
+		public class EtoPanel<THandler> : swf.Panel
+			where THandler: WindowsControl<TControl, TWidget, TCallback>
+		{
+			public THandler Handler { get; }
+
+			public EtoPanel(THandler handler)
+			{
+				Handler = handler;
+				Size = sd.Size.Empty;
+				MinimumSize = sd.Size.Empty;
+				AutoSize = true;
+				AutoSizeMode = swf.AutoSizeMode.GrowAndShrink;
+			}
+
+			public override sd.Size GetPreferredSize(sd.Size proposedSize)
+			{
+				var userSize = Handler.UserDesiredSize;
+				var size = userSize.Width >= 0 && userSize.Height >= 0 ? sd.Size.Empty
+					: base.GetPreferredSize(proposedSize);
```

Comment:
performance improvement, no need to call `base.GetPreferredSize` when we won't use the result